src/eradiate/scenes/surface/_dem.py

```diff
@@ -1,50 +1,320 @@
+    """
+    Generate DEM vertex positions as (latitude, longitude, elevation) pairs
+    for a plane parallel atmosphere geometry.
+
+    Note
+    ----
+    Elevation data is converted into kernel units and the unit is removed.
+
+    Parameters
+    ----------
+    latitude_points : array-like
+        Points along the latitude axis of the DEM
+
+    longitude_points : array-like
+        Points along the longitude axis of the DEM
+
+    elevation : quantity
+        Elevation data in units of length.
+
+    Returns
+    -------
+    vertices : array
+        DEM vertices as (lat, lon, elevation) triplets. Atmospheric geometry
+        is not taken into account at this step
+    """
+    y_gr, x_gr = np.meshgrid(y, x)
+
+    vertices = np.array(
+        (x_gr.ravel(), y_gr.ravel(), elevation.m_as(uck.get("length")).ravel())
+    ).transpose()
+    return vertices
+
+
+def _vertex_index(x, y, len_y):
+    """
+    Vertex index for a given row and column of gridded vertex data
+    This function handles vectorized index lookup with numpy arrays,
+    as well as individual lookup using floats.
+
+    Parameters
+    ----------
+    x : np.array
+        Row index of the vertex grid
+    y : np.array
+        Column index of the vertex grid
+    len_y : float
+        Length of the second dimension of the vertex grid
+
+    Returns
+    -------
+    index : np.array
+        Index of the vertex in a flattened structure.
+    """
+    return x * len_y + y
+
+
+def _generate_face_indices(len_x, len_y):
+    """
+    Generate indices for face definition in a mesh with gridded vertex positions
+
+    Parameters
+    ----------
+
+    len_x : float
+        Length of the row dimension of the vertex grid
+
+    len_y : float
+        Length of the column dimension of the vertex grid
+
+    Returns
+    -------
+
+    face_indices : np.array
+        (n, 3) array defining faces of the triangulated mesh for the DEM.
+    """
+    x = np.array(range(len_x - 1))
+    y = np.array(range(len_y - 1))
+    xg, yg = np.meshgrid(x, y)
+    vertex_sw = _vertex_index(xg.flatten(), yg.flatten(), len_y)
+
+    xg, yg = np.meshgrid(x + 1, y + 1)
+    vertex_ne = _vertex_index(xg.flatten(), yg.flatten(), len_y)
+
+    xg, yg = np.meshgrid(x, y + 1)
+    vertex_nw = _vertex_index(xg.flatten(), yg.flatten(), len_y)
+
+    xg, yg = np.meshgrid(x + 1, y)
+    vertex_se = _vertex_index(xg.flatten(), yg.flatten(), len_y)
+
+    face_indices_1 = np.array((vertex_ne, vertex_sw, vertex_nw)).transpose()
+    face_indices_2 = np.array((vertex_se, vertex_sw, vertex_ne)).transpose()
+    face_indices = np.concatenate((face_indices_1, face_indices_2))
+
+    return face_indices
+
+
+def _transform_vertices_spherical_shell(vertices, planet_radius):
+    """
+    Convert the (lat, lon, elevation) vertices from the initial vertex generation
+    into (x, y, z) values for spherical shell atmosphere geometries.
+    """
+    planet_radius_m = planet_radius.m_as(ucc.get("length"))
```

Comment:
You should require units for the `vertices` argument (and add typing annotations for clarity). As always: units everywhere or nowhere. Then, you can operate safely on
```python
vertices_m = vertices.m_as(length_units)
```